shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -71,18 +96,85 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI path = new AlluxioURI(args[0]);
     int replication = 1;
-    if (cl.hasOption(REPLICATION)) {
-      replication = Integer.parseInt(cl.getOptionValue(REPLICATION));
+    if (cl.hasOption(REPLICATION.getLongOpt())) {
+      replication = Integer.parseInt(cl.getOptionValue(REPLICATION.getLongOpt()));
+    }
+    if (cl.hasOption(THREAD_OPTION.getLongOpt())) {
+      mThreads = Integer.parseInt(cl.getOptionValue(THREAD_OPTION.getLongOpt()));
     }
+    mDistributedLoadServicePool = new ThreadPoolExecutor(mThreads, mThreads, 60,
+        TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+    mDistributedLoadServicePool.allowCoreThreadTimeOut(true);
+    mDistributedLoadService =
+        new ExecutorCompletionService<>(mDistributedLoadServicePool);
     try {
-      load(path, replication);
+      distributedLoad(path, replication);
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       return -1;
     }
     return 0;
   }
 
+  /**
+   * Create a new job to load a file in Alluxio space, makes it resident in memory.
+   *
+   * @param filePath The {@link AlluxioURI} path to load into Alluxio memory
+   * @param replication The replication of file to load into Alluxio memory
+   */
+  private Callable<AlluxioURI> newJob(AlluxioURI filePath, int replication) {
+    return new Callable<AlluxioURI>() {
+      @Override
+      public AlluxioURI call() throws Exception {
+        JobGrpcClientUtils.run(new LoadConfig(filePath.getPath(), replication), 3,
+            mFsContext.getPathConf(filePath));
+        return filePath;
+      }
+    };
+  }
```

Comment:
Instead of making a method, make this an inner class?

Reply:
Good proposal. In the reconsitution the multi-thread CLI, I will make a general class.